base/stl_helpers.hpp

```diff
@@ -45,12 +45,18 @@ struct Comparer<false, T, C>
 
 // Sorts and removes duplicate entries from |v|.
 template <typename T>
-void SortUnique(std::vector<T> & v)
+void SortUnique(vector<T> & v)
 {
   sort(v.begin(), v.end());
   v.erase(unique(v.begin(), v.end()), v.end());
 }
 
+template <typename T, class TFn>
+void EraseIf(vector<T> & v, TFn fn)
```

Comment:
У нас везде уже переход на &&, это критично здесь оставлять по значению? И пускай он дальше копируется, но один раз )

> On Mar 23, 2016, at 23:07, Viktor Govako <notifications@github.com> wrote:
> 
> In base/stl_helpers.hpp <https://github.com/mapsme/omim/pull/2454#discussion_r57227707>:
> 
> >  {
> >    sort(v.begin(), v.end());
> >    v.erase(unique(v.begin(), v.end()), v.end());
> >  }
> >  
> > +template <typename T, class TFn>
> > +void EraseIf(vector<T> & v, TFn fn)
> Функторы в этом контексте обычно lite copy. Если клиенту важно, то он перед вызовом remove_if делал бы что-то типа ref/cref.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2454/files/de3aa465c10122ec1a980ba6162ad33dac4dd2ea#r57227707>